WTG.Analyzers.Test/RuleSetTest.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using NUnit.Framework;
+
+namespace WTG.Analyzers.Test
+{
+	[TestFixture]
+	class RuleSetTest
+	{
+		[Test]
+		public void WarnAll()
+		{
+			var rulesetPath = Path.Combine(TestContext.CurrentContext.TestDirectory, "..", "..", "WTG.Analyzers", "build", "WarnAll.ruleset");
+
+			var ruleSet = XElement.Load(rulesetPath)
+				.Element("Rules")
+				.Elements("Rule")
+				.ToDictionary(
+					x => x.Attribute("Id").Value,
+					x => (DiagnosticSeverity)Enum.Parse(typeof(DiagnosticSeverity), x.Attribute("Action").Value));
+
+			var descriptors = typeof(Rules).GetFields()
+				.Where(x => x.IsStatic && x.IsInitOnly && typeof(DiagnosticDescriptor).IsAssignableFrom(x.FieldType))
+				.Select(x => (DiagnosticDescriptor)x.GetValue(null));
```

Comment:
Is it worth adding an `All` property to the generated code instead of reflecting over it?

Reply:
Not really, maybe if it were used in production code.

Reply:
Or if you test first 😋